pylint/config/_pylint_config/utils.py

```diff
@@ -16,6 +19,39 @@
 SUPPORTED_FORMATS = {"t", "toml", "i", "ini"}
 
 
+class InvalidUserInput(Exception):
+    """Raised whenever a user input is invalid."""
+
+    def __init__(self, message: str, input_value: str, *args: object) -> None:
+        self.message = message
+        self.input = input_value
+        super().__init__(*args)
+
+
+def should_retry_after_invalid_input(
+    func: Callable[[], Any | None]
+) -> Callable[[], None]:
+    """Decorator that handles InvalidUserInput exceptions."""
+
+    def inner_function() -> None:
+        called_once = False
+        while True:
+            try:
+                func()
+            except InvalidUserInput as exc:
+                if called_once and exc.input == "exit()":
+                    print("Stopping 'pylint-config'.")
+                    sys.exit()
+                print(exc.message)
+                print("Type 'exit()' if you want to exit the program.")
```

Comment:
I think the user can exit with ctrl + C maybe we don't have to handle this ?

Reply:
I didn't want to think of the various keybindings windows, Mac and Linux use. And I thought offering a clean exit instead of printing a stracktrace for `KeyboardInterrupt` would be nice.

I'm fine with removing it!